purchase_discount/models/product_supplierinfo.py

```diff
@@ -10,11 +10,13 @@ class ProductSupplierInfo(models.Model):
 
     discount = fields.Float(string="Discount (%)", digits="Discount")
 
-    @api.onchange("name")
-    def onchange_name(self):
+    @api.onchange("partner_id")
+    def onchange_partner_id(self):
```

Comment:
Instead of this, do `discount` a computed writable field depending on `partner_id`:

```python
    discount = fields.Float(string="Discount (%)", digits="Discount", compute="_compute_discount", store=True, readonly=False)

    @api.depends("partner_id")
    def _compute_discount(self):
        for record in self:
            record.discount = record.partner_id.default_supplierinfo_discount
```

Reply:
done